cassiopeia/simulator/LineageTracingDataSimulator.py

```diff
@@ -0,0 +1,25 @@
+"""
+This file stores an abstract subclass of DataSimulator, the
+LineageTracingDataSimulator. A LineageTracingDataSimulator overlays lineage
+tracing data onto a CassiopeiaTree, i.e. it sets the character states of a
+CassiopeiaTree (in particular, the character matrix).
+"""
+import abc
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.simulator.DataSimulator import DataSimulator
+
+
+class LineageTracingDataSimulator(DataSimulator):
+    """
+    LineageTracingDataSimulator is an abstract class that all lineage tracing
+    data simulators derive from.
+    """
+
+    @abc.abstractmethod
+    def overlay_data(self, tree: CassiopeiaTree) -> None:
+        """
+        Overlay lineage tracing data onto the CassiopeiaTree (in-place).
+        This sets the character states of all nodes in the tree, as well
+        as the character matrix.
```

Comment:
Could you please add argument section to documentation?

Also, it would be great to expand a bit more on what we _expect_ trees to have and what they do not have. Should we also have a boolean argument controlling whether or not a user would receive a new tree or whether to perform the operation in place?

Reply:
Thanks, totally forgot about the argument! I added that the trees should have a topology and meaningful branch lengths.

> Should we also have a boolean argument controlling whether or not a user would receive a new tree or whether to perform the operation in place?

I think this makes the interface less clear; all our interfaces currently operate on the tree in place. The users can make a copy of the tree if they want anyway?